backend/inspirehep/records/api/base.py

```diff
@@ -335,9 +335,11 @@ def hard_delete(self):
                 PersistentIdentifier.object_uuid == self.id
             ).all()
             for pid in pids:
-                RecordIdentifier.query.filter_by(recid=pid.pid_value).delete()
+                if pid.pid_provider == "recid":
+                    RecordIdentifier.query.filter_by(recid=pid.pid_value).delete()
```

Comment:
don't you also need to delete the other PIDs?

Reply:
`pid` is not only recid but also doi or arXiv. If you will try to remove them from RecordIdentifier it will blow up as there are only recids.
pid from PersistentIdentifier table should be deleted after this check. (and originally was).

Reply:
Actually it doesn't blow up :)

Reply:
OK, I understand. Was confused between `RecordIdentifier` and `PersistentIdentifier`. I agree it should be there.